prime-router/docs/proposals/0009-proposal-for-history-api.md

```diff
@@ -0,0 +1,232 @@
+```
+The actionParams and actionResult elements are escaped Json/mixed text. Both parameters should be well formatted Json once the prerequisites are implemented. This response is basically serializing a record in the action table to a Json response.
+
+### Report Items
+`GET /report/{reportId}/items?limit=20&offset=0`
+
+This end point will report the status of all items in the given reportId after verifying that the authenticated user has access to the organization that sent the report. This endpoint will also use the same pagination pattern as the endpoint above. 
+
+Exmaple Json:
+```json
+{
+    "result_set": {
+        "count": 20,
+        "offset": 0,
+        "limit": 20,
+        "total": 2
+    },
+    "reportId" : "18a8e461-b3f8-436d-b94e-4065c7b9d186",
+    "createdAt": "2021-06-30 13:03:00.305657-04",
+    "items": [
+        {
+            "trackingId": "reportIndex-0",
+            "destinationCount": 2,
+            "destinations": {
+                "COUNTY": [],
+                "STATE": [
+                    {
+                        "reportId": "33a8e478-b3d2-498d-b94e-4065f7b9e134",
+                        "receiverOrg": "md-doh",
+                        "recieverName": "elr",
+                        "format": "HL7_BATCH",
+                        "sentOn": "2021-06-29 11:00:00.086375-04"
+                    }
+                ],
+                "FEDERAL": [
+                    {
+                        "reportId": "44a8e478-c3f2-498d-b94d-5068f7b9e134",
+                        "receiverOrg": "hhsprotect",
+                        "recieverName": "elr",
+                        "format": "CSV",
+                        "sentOn": "2021-06-29 11:00:00.086375-04"
+                    }
+                ]
+            },
+            "trackingId": "reportIndex-2",
+            "destinationCount": 0,
+            "destinations": {
+                "COUNTY": [],
+                "STATE": [],
+                "FEDERAL": []
+            }
+        }
+    ]
+}
+```
+Alternatively, the jurisdiction breakdown can be removed and added as an element to each individual report. Items routing no where would have a destination count of zero and empty jurisdiction arrays (unless the structure is changed).
+
+Alternative Json Response:
+```json
+{
+    "result_set": {
+        "count": 20,
+        "offset": 0,
+        "limit": 20,
+        "total": 2
+    },
+    "reportId" : "18a8e461-b3f8-436d-b94e-4065c7b9d186",
+    "createdAt": "2021-06-30 13:03:00.305657-04",
+    "items": [
+        {
+            "trackingId": "reportIndex-0",
+            "destinationCount": 1,
+            "destinations": [
+                {
+                    "reportId": "33a8e478-b3d2-498d-b94e-4065f7b9e134",
+                    "receiverOrg": "md-doh",
+                    "recieverName": "elr",
+                    "jurisdiction": "STATE",
+                    "format": "HL7_BATCH",
+                    "sentOn": "2021-06-29 11:00:00.086375-04"
```

Comment:
What you have here is the right approach - we want to provide full lineage history / chain of custody / provenance info.

However, this data is complex to pull out.  _This is not just what's in the Action table_.   in our current json response in the Action table, its called `sending_at` for a reason -- usually, the send is later, and of course it might fail.

You should study the provenance resource in FHIR to see what it wants to provide